indico/modules/events/registration/fields/sessions.py

```diff
@@ -42,18 +55,33 @@ def _check_number_of_sessions(new_data):
                     raise ValidationError(_('Please select no more than {max} sessions.').format(max=_max))
                 if _min != 0 and len(new_data) < _min:
                     raise ValidationError(_('Please select at least {min} sessions.').format(min=_min))
-        return _check_number_of_sessions
+
+        def _check_session_block_is_valid(new_data):
+            event = flask.g.rh.event
```

Comment:
We typically do `from flask import g` (unlike in the UN codebase where AFAIK you always import the module itself).

Reply:
in fact there's an even better solution that avoids using `g` altogether:

```py
event = self.form_item.registration_form.event
```